server/src/internalClusterTest/java/org/elasticsearch/health/HealthMetadataServiceIT.java

```diff
@@ -177,31 +147,27 @@ public void testWatermarkSettingUpdate() throws Exception {
                 var healthMetadata = HealthMetadata.getFromClusterState(internalCluster.clusterService().state());
                 var diskMetadata = healthMetadata.getDiskMetadata();
                 assertThat(diskMetadata.describeHighWatermark(), equalTo(updatedHighWatermark));
-
-                // The value of the setting `cluster.routing.allocation.disk.watermark.high.max_headroom` depends upon the existence of
-                // `cluster.routing.allocation.disk.watermark.high`.
-                // Check {@link DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_MAX_HEADROOM_SETTING}
-                assertThat(
-                    diskMetadata.highMaxHeadroom(),
-                    equalTo(
-                        percentageMode
-                            ? updatedHighMaxHeadroom
-                            : CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_MAX_HEADROOM_SETTING.getDefault(EMPTY)
-                    )
-                );
+                assertThat(diskMetadata.highMaxHeadroom(), equalTo(updatedHighMaxHeadroom));
                 assertThat(diskMetadata.describeFloodStageWatermark(), equalTo(updatedFloodStageWatermark));
+                assertThat(diskMetadata.floodStageMaxHeadroom(), equalTo(updatedFloodStageMaxHeadroom));
+
+                var shardLimitsMetadata = healthMetadata.getShardLimitsMetadata();
+                assertEquals(shardLimitsMetadata, updatedShardLimits);
+            });
+
+            var electedMaster = internalCluster.getMasterName();
 
-                // The value of the setting `cluster.routing.allocation.disk.watermark.flood_stage.max_headroom` depends upon the existence
-                // of `cluster.routing.allocation.disk.watermark.flood_stage`.
-                // Check{@link DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_MAX_HEADROOM_SETTING}
-                assertThat(
-                    diskMetadata.floodStageMaxHeadroom(),
-                    equalTo(
-                        percentageMode
-                            ? updatedFloodStageMaxHeadroom
-                            : CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_MAX_HEADROOM_SETTING.getDefault(EMPTY)
-                    )
-                );
+            // Force a master fail-over but, since the settings were manually changed, we should return the manually set values
+            internalCluster.stopNode(electedMaster);
+            ensureStableCluster(numberOfNodes - 1);
+
+            assertBusy(() -> {
+                var healthMetadata = HealthMetadata.getFromClusterState(internalCluster.clusterService().state());
```

Comment:
```suggestion
                var healthMetadata = HealthMetadata.getFromClusterState(internalCluster.clusterService(internalCluster.getMasterName()).state());
```

Reply:
Let's avoid potential flakiness by always reading the state from the master node